pkg/provider/gitlab/gitlab.go

```diff
@@ -214,26 +243,48 @@ func (g *Gitlab) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretData
 	// 	"masked": true,
 	// 	"environment_scope": "*"
 	// }
-
 	var vopts *gitlab.GetProjectVariableOptions
 	if g.environment != "" {
 		vopts = &gitlab.GetProjectVariableOptions{Filter: &gitlab.VariableFilter{EnvironmentScope: g.environment}}
 	}
-	data, _, err := g.client.GetVariable(g.projectID, ref.Key, vopts)
-	if err != nil {
+
+	data, resp, err := g.projectClient.GetVariable(g.projectID, ref.Key, vopts)
+	if resp.StatusCode >= 400 && err != nil {
 		return nil, err
 	}
 
+	var result []byte
+	result, err = extractVariable(ref, data.Value)
+	for i := len(g.groupIDs) - 1; i >= 0; i-- {
+		groupID := g.groupIDs[i]
+		if result != nil {
+			return result, nil
+		}
+
+		groupVar, resp, err := g.groupClient.GetVariable(groupID, ref.Key, nil)
+		if resp.StatusCode >= 400 && err != nil {
+			return nil, err
+		}
+		result, _ = extractVariable(ref, groupVar.Value)
```

Comment:
we shouldn't ignore the error here.

Reply:
In case there is no result in any of the groups it will use the error returned in line 257, stating that the secret was not found.